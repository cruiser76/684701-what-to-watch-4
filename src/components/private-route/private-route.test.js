import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {MemoryRouter} from "react-router-dom";

import PrivateRoute from './private-route.jsx';

const mockStore = configureStore([]);
const store = mockStore({
  USER: {authorizationStatus: `NO_AUTH`}
});

describe(`Render correct PrivateRoute`, () => {
  it(`Render PrivateRoute`, () => {
    const tree = renderer
    .create(
        <Provider store={store}>
          <MemoryRouter>
            <PrivateRoute
              render={() => {}}
              path={`/`}
              exact={true}
              authorizationStatus={`AUTH`}
            />
          </MemoryRouter>
        </Provider>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
