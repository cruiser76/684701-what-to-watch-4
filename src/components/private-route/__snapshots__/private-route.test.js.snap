// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correct PrivateRoute Render PrivateRoute 1`] = `null`;
